salt/config/__init__.py

```diff
@@ -3273,10 +3273,11 @@ def api_config(path):
     '''
     # Let's grab a copy of salt's master default opts
     defaults = DEFAULT_MASTER_OPTS
+    opts = client_config(path, defaults=defaults)
     # Let's override them with salt-api's required defaults
-    defaults.update(DEFAULT_API_OPTS)
+    opts.update(DEFAULT_API_OPTS)
```

Comment:
@s0undt3ch If I set the "pidfile" and "log_file" in /etc/salt/master, the salt-api will use the same pid and log_file as salt-master too. Their logs  were mixed up and pidfile were covered by each other. Using the default may not be the best solution. Maybe we should use api_log_file and api_pidfile, just like syndic_log_file and syndic_pidfile